### PR TITLE
Stop mocking `DomainSerdes`

### LINKED ISSUE #1394: Eliminate mocking of `DomainSerdes`
**Summary**
Our unit test strategy for `SelfSerializable` types has suffered from undue mocking; a special pain point is mocking of static instances of the `DomainSerdes` utility, since with concurrent unit test execution, a test attempting a "live-fire" serialization/deserialization may fail by hitting a mock!

**Suggested resolution**
For every `SelfSerializable` type that uses `DomainSerdes`, replace mock-dependent unit test coverage by switching to a `SelfSerializableDataTest`.

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
- Refines `SelfSerializableDataTest` by integrating with a [new `SerializedForms` helper](https://github.com/hashgraph/hedera-services/blob/01394-stop-mocking-DomainSerdes/hedera-node/src/test/java/com/hedera/test/serde/SerializedForms.java#L64) that stores instances created from a `SeededPropertySource` in this [resources directory.](https://github.com/hashgraph/hedera-services/tree/01394-stop-mocking-DomainSerdes/hedera-node/src/test/resources/serdes)
- Replaces all unit tests that required mocking a static `DomainSerdes` instance with a `SelfSerializableDataTest`.

**Related issue(s)**:
- Fixes #1394

### CODE REVIEW

**Review by rjuang — CHANGES_REQUESTED** [review]

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleTopic.java [hunk 1]
```diff
@@ -384,6 +401,10 @@ public RichInstant getExpirationTimestamp() {
 		return hasExpirationTimestamp() ? expirationTimestamp : new RichInstant();
 	}
 
+	public RichInstant getNullableExpirationTimestamp() {
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleTopic.java:400]: Done.

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java [hunk 2]
```diff
@@ -92,19 +88,19 @@
 	private boolean migrationRecordsStreamed;
 	private FeeMultiplierSource multiplierSource = null;
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java:90]: Done.

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java [hunk 3]
```diff
@@ -55,45 +55,38 @@
...
 
-	public static final Key UNUSED_GRPC_KEY = null;
-	public static final JKey UNUSED_KEY = null;
-	public static final EntityId UNUSED_PAYER = null;
-	public static final RichInstant UNRESOLVED_TIME = null;
+	static final int MAX_NUM_PUBKEY_BYTES = 33;
 
-	private Key grpcAdminKey = UNUSED_GRPC_KEY;
-	private JKey adminKey = UNUSED_KEY;
+	private Key grpcAdminKey = null;
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java:73]: Done.

File: hedera-node/src/main/java/com/hedera/services/state/serdes/IoUtils.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+	public static <T> T staticReadNullable(
+			final SerializableDataInputStream in,
+			final IoReadingFunction<T> reader
+	) throws IOException {
+		return in.readBoolean() ? reader.read(in) : null;
+	}
+
+	public static <T extends SelfSerializable> void staticWriteNullableSerializable(
+			final T data,
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/state/serdes/IoUtils.java]: Done.

File: hedera-node/src/main/java/com/hedera/services/state/serdes/IoUtils.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+public class IoUtils {
+	public static void serializeKey(final JKey key, final DataOutputStream out) throws IOException {
+		out.write(key.serialize());
+	}
+
+	@Nullable
```
Comment (rjuang) [inline hedera-node/src/main/java/com/hedera/services/state/serdes/IoUtils.java:39]: The nullable needs to be documented. What is written if the value is null and when is null returned ?

**Review by rjuang — APPROVED** [review]

**Pull request merged**